client/my-sites/posts/post.jsx

```diff
@@ -46,32 +48,46 @@ function checkPropsChange( currentProps, nextProps, propArr ) {
 }
 
 const Post = React.createClass( {
-
 	displayName: 'Post',
 
-	mixins: [ updatePostStatus ],
+	propTypes: {
+		// connected via updatePostStatus
+		buildUpdateTemplate: PropTypes.func.isRequired,
+		togglePageActions: PropTypes.func.isRequired,
+		updatePostStatus: PropTypes.func.isRequired,
+		updated: PropTypes.bool.isRequired,
+		updatedStatus: PropTypes.string,
+		previousStatus: PropTypes.string,
+		showMoreOptions: PropTypes.bool.isRequired,
+		showPageActions: PropTypes.bool.isRequired,
+	},
 
 	getInitialState() {
 		return {
-			showMoreOptions: false,
 			showComments: false,
 			showShare: false,
 			commentsFilter: 'all'
 		};
 	},
 
 	shouldComponentUpdate( nextProps, nextState ) {
-		const propsToCheck = [ 'post', 'postImages', 'fullWidthPost', 'path' ];
-
-		if ( checkPropsChange( this.props, nextProps, propsToCheck ) ) {
-			return true;
-		}
-
-		if ( nextState.showMoreOptions !== this.props.showMoreOptions ) {
+		if ( ! shallowEqual( this.state, nextState ) ) {
 			return true;
 		}
 
-		return false;
+		const propsToCheck = [
+			'post',
+			'postImages',
+			'fullWidthPost',
+			'path',
+
+			// via updatePostStatus
+			'previousStatus',
+			'showMoreOptions',
+			'updated',
+			'updatedStatus',
+		];
+		return checkPropsChange( this.props, nextProps, propsToCheck );
```

Comment:
It's just shallow equality. The first part of the method only cares about `state` and is more defensive: if _any_ state has changed, then update. For the second part, we pick designated bits of `props`. I'm only following the pattern that was already in place, and added the new props coming from `updatePostStatus` that I deem necessary to listen to.

My guess is that `checkPropsChange` was originally introduced to cope with the fact that pre-Redux _a lot_ of props were passed to Post — some of which likely mutable, some immutable — and so the component only compared a handful to determine whether to update.

Also, I added the bit about comparing `state` because, without it, some actions like toggling the comments sub-panel weren't triggering re-renders at all.